### PR TITLE
Added new missed state for alarms (#420)

### PR DESCRIPTION
**Pull request by williamg97**
Added status 5 which seems to be missed status. I was able to narrow it down to which of my many devices was causing it and then could see this alarm missed alert when clicking on the alarm icon. Once I'd cleared the following it started to work again. 

image

### CODE REVIEW

File: custom_components/google_home/sensor.py [hunk 1]
```diff
@@ -179,7 +179,9 @@ def state(self) -> str | None:
         next_alarm = device.get_next_alarm()
         return (
             next_alarm.local_time_iso
-            if next_alarm and next_alarm.status != GoogleHomeAlarmStatus.INACTIVE
+            if next_alarm
+            and next_alarm.status != GoogleHomeAlarmStatus.INACTIVE
+            and next_alarm.status != GoogleHomeAlarmStatus.MISSED
```
Comment (leikoilja) [inline custom_components/google_home/sensor.py:184]: How about a shorthand version here to keep the implementation consistent?

```suggestion
            and next_alarm.status not in (GoogleHomeAlarmStatus.INACTIVE, GoogleHomeAlarmStatus.MISSED)
```

**Review by leikoilja — APPROVED** [review]
Amazing! Thanks, @williamg97 for digging out and finding what missing status 5 means :)

File: custom_components/google_home/sensor.py [hunk 2]
```diff
@@ -179,7 +179,8 @@ def state(self) -> str | None:
         next_alarm = device.get_next_alarm()
         return (
             next_alarm.local_time_iso
-            if next_alarm and next_alarm.status != GoogleHomeAlarmStatus.INACTIVE
+            if next_alarm
+            and next_alarm.status not in (GoogleHomeAlarmStatus.INACTIVE, GoogleHomeAlarmStatus.MISSED)
```
Comment (DurgNomis-drol) [inline custom_components/google_home/sensor.py:183]:
```suggestion
            and next_alarm.status
            not in (GoogleHomeAlarmStatus.INACTIVE, GoogleHomeAlarmStatus.MISSED)
```

**Review by DurgNomis-drol — CHANGES_REQUESTED** [review]

**Review by leikoilja — APPROVED** [review]

**Comment by leikoilja** [comment]
Awesome find, @williamg97! Thanks for fixing it 🚀 🔥

**Pull request merged**